cli/toolchain_cmd.go

```diff
@@ -663,3 +669,20 @@ func symlink(source, target string) error {
 		return err
 	}
 }
+
+// prepareParentDir makes parent directory for a given toolchain under SRCLIBPATH and returns SRCLIBPATH/FULL-PATH-TO-TOOLCHAIN
+func prepareParentDir(toolchainName string) (toolchainDir string, err error) {
+	// toolchain dir under SRCLIBPATH
+	srclibpathDir := filepath.Join(filepath.SplitList(srclib.Path)[0], toolchainFqn(toolchainName))
+	return srclibpathDir, os.MkdirAll(filepath.Dir(srclibpathDir), 0700)
+}
+
+// toolchainFqn returns fully qualified toolchain name
```

Comment:
Go style says that acronyms should be all upper case, see https://github.com/golang/go/wiki/CodeReviewComments#initialisms.